src/System.Numerics.Tensors/src/System/Numerics/Tensors/ITensor.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Numerics.Tensors
+{
+    public interface ITensor<T>
+    {
+        ReadOnlySpan<int> Dimensions { get; }
+        bool IsReversedStride { get; }
+        long Length { get; }
+        int Rank { get; }
+        ReadOnlySpan<int> Strides { get; }
+
+        T this[params int[] indices] { get; set; }
+        T this[ReadOnlySpan<int> indices] { get; set; }
+
+        T GetValue(int index);
+        void SetValue(int index, T value);
+
+        ITensor<T> Clone();
```

Comment:
I'm not convinced `Clone` or `CloneEmpty` are needed as part of the interface....

Reply:
If we want to be able to construct new tensors of the same type we need some way to do that.  The implementations of arithmetic, getdiagonal, etc, needed to do that.

Reply:
Does that need to be something that is part of the `Tensor` contract however or should it rather be something that is done and exposed only by the concrete types?

Reply:
I don't like the names for certain and it doesn't feel like something that should be part of this interface so I can agree that we should remove Clone and CloneEmpty.

The scenario is that someone wants to write code that handles tensors of any type and does something to produce a new tensor..  Maybe we'd be OK with that code choosing its own mechanism for producing the new tensor, for example it always returns a DenseTensor.  The downside of that is what if you wanted to turn around and pass that output to some DNN library and it needed a tensor in it's own memory?  We need to come up with an answer for this scenario and make sure we have the API to support it.

Reply:
I'll jot this down as an open question for us to follow up on.